.gitlab-ci.yml

```
@@ -25,25 +26,49 @@ variables:
   - &HELMUPGRADE helm --tiller-namespace ${KUBE_NAMESPACE} upgrade --install --force "${DEPLOYMENT_NAME}" etc/helm --set hostEnvVar="${CASA_SERVICE}HOST" --set authEnvVar="${CASA_SERVICE}PORT_AUTHORIZED_PORT" --set portEnvVar="${CASA_SERVICE}PORT_PORT" --set name="${DEPLOYMENT_NAME}" --set app="${DEPLOYMENT_APP}" --set image.image="${DEPLOYMENT_IMAGE}" --values etc/helm/values/$CI_ENVIRONMENT_NAME.yaml --namespace ${KUBE_NAMESPACE}
   - &HELMCHECK helm --tiller-namespace ${KUBE_NAMESPACE} ls &&
     helm --tiller-namespace ${KUBE_NAMESPACE} status "${DEPLOYMENT_NAME}"
+  - &ZIPFILES bzip2 -zk etc/docker/_artifacts/* &&
+    mkdir -p etc/docker/_artifacts/bz2 &&
+    mv etc/docker/_artifacts/*.bz2 etc/docker/_artifacts/bz2
 
 build:
   stage: build
   script:
     # Clear *_TOKEN variables during code build so that compile-time code can't access them
-    - apt-get update && apt-get install -y --no-install-recommends libpq-dev git rsync
-    - CI_BUILD_TOKEN="" KUBE_TOKEN="" PROD_KUBE_TOKEN="" PROD_DOCKER_PASSWORD="" etc/scripts/stage_docker.sh --install-ghc --test
-    - docker build -t "${DEPLOYMENT_IMAGE}" etc/docker
-    - docker login -u gitlab-ci-token -p "${CI_BUILD_TOKEN}" "${CI_REGISTRY}"
-    - docker push "${DEPLOYMENT_IMAGE}"
-    - |
-      if [[ "$CI_BUILD_REF_NAME" == "master" ]]; then
-        docker tag "${DEPLOYMENT_IMAGE}" "${CI_REGISTRY_IMAGE}:latest"
-        docker push "${CI_REGISTRY_IMAGE}:latest"
-      fi
-      if [[ "$CI_BUILD_REF_NAME" == "ci-cron" ]]; then
-        docker tag "${DEPLOYMENT_IMAGE}" "${CI_REGISTRY_IMAGE}:ci-cron"
-        docker push "${CI_REGISTRY_IMAGE}:ci-cron"
-      fi
+    - cache-s3 restore stack
+    - cache-s3 restore stack work
+    - etc/scripts/stage_docker.sh --install-ghc --test
+    - cache-s3 save stack
+    - cache-s3 save stack work
+  artifacts:
+    name: "${CI_BUILD_REF_SLUG}_${CI_PIPELINE_ID}"
+    paths:
+      - etc/docker/_artifacts/
+
+deploy_prod_s3:
+  stage: deploy
+  only:
+    - master
+  environment:
+    name: production
+  before_script:
+    - *ZIPFILES
+  script:
+    - aws s3 cp --acl=public-read etc/docker/_artifacts/bz2/curator.bz2 s3://${CI_S3_BUCKET}/${CI_S3_DIRECTORY}/curator-latest-linux-amd64.bz2
+    - aws s3 cp --acl=public-read etc/docker/_artifacts/bz2/casa-curator.bz2 s3://${CI_S3_BUCKET}/${CI_S3_DIRECTORY}/casa-curator-latest-linux-amd64.bz2
+    - aws s3 cp --acl=public-read etc/docker/_artifacts/bz2/curator.bz2 s3://${CI_S3_BUCKET}/${CI_S3_DIRECTORY}/curator-${CI_BUILD_TAG}-linux-amd64.bz2
+    - aws s3 cp --acl=public-read etc/docker/_artifacts/bz2/casa-curator.bz2 s3://${CI_S3_BUCKET}/${CI_S3_DIRECTORY}/casa-curator-${CI_BUILD_TAG}-linux-amd64.bz2
```

```suggestion
    - aws s3 cp --acl=public-read etc/docker/_artifacts/bz2/casa-curator.bz2 s3://${CI_S3_BUCKET}/${CI_S3_DIRECTORY}/casa-curator-${CI_COMMIT_TAG}-linux-amd64.bz2
```

`CI_BUILD_TAG` has been deprecated in favor of `CI_COMMIT_TAG`

Link for reference: https://docs.gitlab.com/ee/ci/variables/deprecated_variables.html

---

Also from the docs for `CI_COMMIT_TAG`:

> The commit tag name. Present only when building tags.

Is that going to be an issue? This job will only run on master. Are there situations where this env var will be empty though?